parsing/parser.mly

```diff
@@ -2467,6 +2467,9 @@ labeled_simple_expr:
   | TILDE label = LIDENT
       { let loc = $loc(label) in
         (Labelled label, mkexpvar ~loc label) }
+  | TILDE LPAREN label = LIDENT ty = type_constraint RPAREN
+      { let loc = $loc(label) in
+        (Labelled label, mkexp_constraint ~loc (mkexpvar ~loc label) ty) }
```

Comment:
I wouldn't put the location of the label for the constraint expression, but `~loc:($startpos($2), $endpos)` instead.

For reference, the parse tree of `(0 : int)` (in the toplevel) is : 
```
Ptop_def
  [
    structure_item (//toplevel//[1,0+0]..[1,0+9]) ghost
      Pstr_eval
      expression (//toplevel//[1,0+0]..[1,0+9]) ghost
        Pexp_constraint
        expression (//toplevel//[1,0+1]..[1,0+2])
          Pexp_constant PConst_int (1,None)
        core_type (//toplevel//[1,0+5]..[1,0+8])
          Ptyp_constr "int" (//toplevel//[1,0+5]..[1,0+8])
          []
  ]
```
where the constraint location encapsulates everything.

Reply:
Fixed, thank you.